tests/modules/redshift_datasets/conftest.py

```diff
@@ -54,25 +57,34 @@ def get_workgroup_arn(self, *args, **kwargs):
         return 'arn:aws:redshift-serverless:eu-west-1:XXXXXXXXXXXXXX:workgroup/workgroup_name_1'
 
 
-@pytest.fixture(scope='function')
-def patch_redshift(module_mocker):
-    # Mocking connection service
+@pytest.fixture(scope='module', autouse=True)
+def patch_sts_remote_session(module_mocker):
     module_mocker.patch(
-        'dataall.modules.redshift_datasets.services.redshift_connection_service.redshift_client',
+        'dataall.base.aws.sts.SessionHelper.remote_session',
+        return_value=boto3.Session(),
+    )
+
+
+@pytest.fixture(scope='function')
+def patch_redshift(mocker):
+    # autospec=True ensures methods called in the MockClient correspond to real client methods
+    mocker.patch.object(
```

Comment:
I don't think you need the Mock classes at all, take a look at the following example...

```python
from unittest.mock import NonCallableMagicMock

import pytest


class ARealClass:
    def get_str(self):
        return 'real_str'

    @staticmethod
    def static_func():
        return 'static_str'


def test_patch(mocker):
    mock_str = '123'
    mock_class = mocker.patch('scratch.ARealClass', autospec=True)
    mock_class.static_func.return_value = mock_str
    mock_class().get_str.return_value = mock_str

    # check that static function was mocked
    assert ARealClass.static_func() == mock_str
    # check that creating ARealClass() will now created a MagicMock
    cls_instance = ARealClass()
    assert isinstance(cls_instance, NonCallableMagicMock)
    # check that get_str return the mock_str and not the real_str
    assert cls_instance.get_str() == mock_str
    # check that autospec is working and calling arbitrary methods throws
    with pytest.raises(AttributeError):
        ARealClass().non_existend_method()
```